servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -2820,7 +2820,20 @@ Role updateTemplateRole(Role role, String domainName, List<TemplateParam> params
         }
         Role templateRole = new Role()
```

Comment:
The change itself is not sufficient since it's possible that the user has made some changes already and thus we don't want to lose any of those changes. So I want to make our logic a little more smarter.

When applying the template and updating the role, we should read the original values and only override those that have been specified in the template. (let's have a unit test for this case as well).

Reply:
Done. Added logic to merge existing rolemeta from DB with new rolemeta values form template.
added a unit test to test this scenario too.